api/proto/teleport/legacy/types/types.proto

```diff
@@ -3926,6 +3926,21 @@ message OIDCConnectorSpecV3 {
   bool AllowUnverifiedEmail = 15 [(gogoproto.jsontag) = "allow_unverified_email,omitempty"];
   // UsernameClaim specifies the name of the claim from the OIDC connector to be used as the user's username.
   string UsernameClaim = 16 [(gogoproto.jsontag) = "username_claim,omitempty"];
+  // MaxAge is the amount of time that user logins are
+  // valid for. If a user logs in, but then does not login again
+  // within this time period, they will be forced to re-authenticate.
+  MaxAge MaxAge = 17 [
+    (gogoproto.jsontag) = "",
+    (gogoproto.embed) = true
+  ];
+}
+
+// MaxAge allows the max_age parameter to be unset to preserve backwards compatibility.
+message MaxAge {
+  int64 Value = 1 [
+    (gogoproto.jsontag) = "max_age",
+    (gogoproto.casttype) = "Duration"
+  ];
```

Comment:
Given that the field is a duration why don't you use the well known types:

```suggestion
  google.protobuf.Duration MaxAge = 17 [
    (gogoproto.jsontag) = "max_age",
    (gogoproto.stdduration) = true
  ];
}
```

This is compiled to:

```go
MaxAge *time.Duration
```

This also fixes your operator issues

Reply:
and the operator transforms it to `string` instead of `int32`

```
              max_age:
                description: MaxAge is the amount of time that user logins are valid
                  for. If a user logs in, but then does not login again within this
                  time period, they will be forced to re-authenticate.
                format: duration
                type: string
```
---

**Having a duration with `int32` is a HUGE problem because int32 can only encode durations up to `2.14s`**

Reply:
I'm weary of using `(gogoproto.stdduration) = true` instead of `(gogoproto.casttype) = "Duration"`, as all existing code uses `Duration`. I'm not 100% sure why, but I'm sure there's a good reason for it. Maybe the CRD just needs to be updated to handle this case?